lib/Pusher.php

```diff
@@ -184,6 +184,37 @@ private function check_compatibility()
 		}
 
 	}
+
+	/**
+	 * validate number of channels and channel name format.
+	 */	
+	private function validate_channels($channels) {
+		if( count( $channels ) > 100 ) {
+			throw new PusherException('An event can be triggered on a maximum of 100 channels in a single call.');
+		}
+		
+		array_walk( $channels, array( $this, 'validate_channel' ) );
+	}
+
+	/**
+	 * Ensure a channel name is valid based on our spec
+	 */
+	private function validate_channel( $channel )
+	{
+		if ( ! preg_match( '/\A[-a-zA-Z0-9_=@,.;]+\z/', $channel ) ) {
+			throw new PusherException( 'Invalid channel name ' . $channel );
+		}
+	}
+
+	/**
+	 * Ensure a socket_id is valid based on our spec
+	 */
+	private function validate_socket_id( $socket_id )
+	{
+		if ( $socket_id && !preg_match( '/\A\d+\.\d+\z/', $socket_id ) ) {
```

Comment:
The way I see it is that we want a valid socket_id to either be `null` (the default function parameter value), or a string which matches the regex. Anything else should fail, including all falsey values e.g. false, array(), 0, and the important one: "".

The `preg_match` is OK because from my tests it appears to return false when matching these values, however the current check for $socket_id does not check that it's anything except null, it checks that it's any truthy value, therefore the falsey values will not pass this check, and therefore the `preg_match` will never be executed on them, and the exception will not be thrown.

I don't think we need to check it's type string, because the `preg_match` should cover this.

I can add tests to illustrate this.